api/src/infrastructure/docker.rs

```diff
@@ -499,62 +556,119 @@ impl Infrastructure for DockerInfrastructure {
 
     async fn deploy_services(
         &self,
+        deployment_id: &String,
         app_name: &String,
         configs: &Vec<ServiceConfig>,
         container_config: &ContainerConfig,
     ) -> Result<Vec<Service>, Error> {
-        let network_id = self.create_or_get_network_id(app_name).await?;
+        let deployment_container = self
+            .create_deployment_task_container(deployment_id, app_name)
+            .await?;
 
-        self.connect_traefik(&network_id).await?;
+        let result = (async move {
```

Comment:
You need to help me here: why it is necessary to wrap the code into a `async move` block?

Reply:
Maybe there's a better way, but the part inside the block handles a bunch of error conditions with the `?` operator.
I wanted to make sure that the temporary container gets deleted even in cases where the function exits early because of an error. I looked at implementing the `Drop` trait or using something like the `scopeguard` crate to take care of that. However, since `delete` is async, both of those don't work (unless find a way to poll the completion synchronously).